### PR TITLE
Update lib_webgpu_dawn from Unity changes

### PR DESCRIPTION
**Pull request by brendan-duncan**
Update the main repo with the changes made to lib_webgpu_dawn from Unity.

### CODE REVIEW

File: lib/lib_webgpu_dawn.cpp [hunk 1]
```diff
@@ -622,8 +623,19 @@ WGPU_DEVICE_LOST_REASON Dawn_to_WGPU_DEVICE_LOST_REASON[] = {
 
 //////////////////////////////////////////////////////////////////////
 
+#ifdef DAWN_UNSAFE_API
+const char* enabledToggles = "allow_unsafe_apis";
+
+WGPUDawnTogglesDescriptor instanceTogglesDesc{ { nullptr, WGPUSType_DawnTogglesDescriptor}, 1, &enabledToggles };
+WGPUInstanceDescriptor instanceDesc{ reinterpret_cast<WGPUChainedStruct*>(&instanceTogglesDesc) };
```
Comment (juj) [inline lib/lib_webgpu_dawn.cpp:630]: Can you add `static` to the above three global objects so they retain .cpp local linkage?

File: lib/lib_webgpu_dawn.cpp [hunk 2]
```diff
@@ -1756,7 +1775,36 @@ void wgpu_buffer_map_async(WGpuBuffer buffer, WGpuBufferMapCallback callback, vo
...
+  
+  auto callback = [](WGPUBufferMapAsyncStatus status, void* rawData) {
+  	auto* userdata = static_cast<_Data*>(rawData);
+  	userdata->done = true;
+  	_Data* data = (_Data*)userdata;
+  	_WGpuObjectBuffer* obj = (_WGpuObjectBuffer*)_wgpu_get(data->buffer);
+  	obj->state = (data->mode & WGPUMapMode_Write) ? kWebGPUBufferMapStateMappedForWriting : kWebGPUBufferMapStateMappedForReading;
+  };
+  
+  _Data* data = new _Data{ buffer, mode };
```
Comment (juj) [inline lib/lib_webgpu_dawn.cpp:1800]: Does this need to be allocated with `new`? Or would a stack allocated `_Data data = { buffer, mode };` work? (good either way, just curious)

**Comment by juj** [comment]
LGTM, feel free to revise in a followup PR if you'd like.

File: lib/lib_webgpu_dawn.cpp (hunk 2 above)
Comment (brendan-duncan) [reply lib/lib_webgpu_dawn.cpp:1800]: I suppose this particular one could be a local var. It's a very low frequency function, so I don't know if putting it on the stack would save much.

**Pull request merged**